lib/MusicBrainz/Server/Edit/Role/IPI.pm

```diff
@@ -12,6 +23,62 @@ with 'MusicBrainz::Server::Edit::Role::ValueSet' => {
     extract_value => sub { shift->ipi }
 };
 
+after initialize => sub {
+    my ($self, %opts) = @_;
+
+    my $old_ipis = $self->data->{old}{ipi_codes};
+    my $new_ipis = $self->data->{new}{ipi_codes};
+    my $added_ipis_set = Set::Scalar->new(@$new_ipis) - Set::Scalar->new(@$old_ipis);
+    my @added_ipis = $added_ipis_set->members;
+    $self->reused_ipis($self->c->model($self->_edit_model)->find_reused_ipis(@added_ipis));
+};
+
+after post_insert => sub {
+    my $self = shift;
+
+    if ($self->reused_ipis) {
+        for my $ipi (keys %{ $self->reused_ipis }) {
```

Comment:
That will always be true, but you could do
```suggestion
    my %reused_ipis = %{$self->reused_ipis};
    if (%reused_ipis) {
        for my $ipi (keys %reused_ipis) {
```
or probably just remove the `if` since the for loop will already be a no-op if `keys %reused_ipis` is empty